wallet-new/integration/Functions.hs

```diff
@@ -677,25 +700,25 @@ chooseActions n probs = liftIO . generate $ do
 respToRes
     :: forall m a. (MonadThrow m)
     => m (Either ClientError (WalletResponse a))
-    -> m a
+    -> PropertyM m a
 respToRes resp = do
-    result <- resp
-    either throwM (pure . wrData) result
+    result <- lift resp
+    run $ either Universum.throwM (pure . wrData) result
 
 
 -- | Pick a random element using @IO@.
-pickRandomElement :: (MonadIO m, Alternative m) => [a] -> m a
-pickRandomElement [] = empty
-pickRandomElement xs = liftIO . generate . elements $ xs
+pickRandomElement :: (Show a, MonadIO m) => [a] -> PropertyM m a
+pickRandomElement [] = liftIO empty
+pickRandomElement xs = pick . elements $ xs
 
 -- | A util function for checking the validity of invariants.
 checkInvariant
-    :: forall m. (MonadThrow m)
+    :: forall m. (MonadIO m)
     => Bool
     -> WalletTestError
-    -> m ()
+    -> PropertyM m ()
 checkInvariant True  _             = pure ()
-checkInvariant False walletTestErr = throwM walletTestErr
+checkInvariant False walletTestErr = liftIO $ Universum.throwM walletTestErr
```

Comment:
Since `PropertyM` has a `MonadThrow` instance (defined above), we can avoid these changes.

Reply:
indeed!!!